interface/controller/item_controller.go

```diff
@@ -0,0 +1,54 @@
+package controller
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"academy_bootcamp/application/usecase"
+	"academy_bootcamp/domain/model"
+)
+
+type itemController struct {
+	Usecase usecase.ItemService
+}
+
+type ItemController interface {
+	GetItemById(c Context) error
+	// CreateItem(c Context) error
+}
+
+func NewItemController(us usecase.ItemService) ItemController {
+	return &itemController{us}
+}
+
+func (uc *itemController) GetItemById(c Context) error {
+	var item *model.Item
+	key, err := strconv.ParseUint(c.Param("id"), 10, 16)
+	if err != nil {
+		return errors.New("QueryParam invalido")
```

Comment:
When the conversion of query param `id` fails it just returns an Internal Server Error message. You could create your own project errors definition and display a little more information to let the user know what's going on.